src/Sylius/Bundle/ApiBundle/ApiPlatform/ConfigMergeManager.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\ApiPlatform;
+
+/** @experimental */
+class ConfigMergeManager
```

Comment:
Making this class final will not change anything. What is more, we should provide an interface to provide proper contract and future possibilities to replace this implementation with something else 
```suggestion
final class ApiResourceConfigurationMerger implements ApiResourceConfigurationMergerInterface
```